raiden/network/proxies/token_network.py

```diff
@@ -2227,19 +2227,83 @@ def settle(
+                                                   error_prefix,
+                                                   log_details,
+                                                   partner,
+                                                   transaction_error,
+                                                   False,
+                                                   gas_limit)
+
+        log.info("settle successful", **log_details)
+
+    def settle_light(
+        self,
+        given_block_identifier: BlockSpecification,
+        channel_identifier: ChannelID,
+        creator: Address,
+        partner: Address,
+        signed_settle_tx: SignedTransaction
+    ):
+        """ Send a signed settle transaction for a specific channel """
+        log_details = {
+            "given_block_identifier": given_block_identifier.hex(),
+            "channel_identifier": channel_identifier,
+            "creator": pex(creator),
+            "partner": pex(partner),
+        }
+        log.debug("settle light called", **log_details)
+
+        checking_block = self.client.get_checking_block()
+        try:
+            self._settle_preconditions(
+                channel_identifier=channel_identifier,
+                partner=partner,
+                block_identifier=given_block_identifier,
+            )
+        except NoStateForBlockIdentifier:
+            # If preconditions end up being on pruned state skip them. Estimate
+            # gas will stop us from sending a transaction that will fail
+            pass
+
+        with self.channel_operations_lock[partner]:
+            transaction_hash = self.proxy.broadcast_signed_transaction(signed_settle_tx)
+            self.client.poll(transaction_hash)
+            transaction_error = check_transaction_threw(self.client, transaction_hash)
+
+        self.validate_settlement_transaction_error(channel_identifier,
+                                                   checking_block,
+                                                   "settle call failed",
+                                                   log_details,
+                                                   partner,
+                                                   transaction_error,
+                                                   True)
+
+        log.info("settle light successful", **log_details)
+
+    def validate_settlement_transaction_error(self,
+                                              channel_identifier,
+                                              checking_block,
+                                              error_prefix,
+                                              log_details,
+                                              partner,
+                                              transaction_error,
+                                              is_light,
+                                              gas_limit=None):
+        transaction_executed = is_light or gas_limit is not None
```

Comment:
I have the same question here, if light_client is true then the transaction executed? the tx for light clients can fail too

Reply:
This was made as a requested change for generalize this, the original code was worse than this, basically if light_client is true i don't have a gas limit to check so i can asume that the transaction was executed but, if you see the next condition you will understand that you can have a transaction_error so basically it will control errors for both cases, i know the conditions are not the best but it's the only way to generalize this error handling, otherwise we have to have the old code here

Reply:
i understand we are trying to make an improvement, but i'm still having trouble understanding this code.

i see this function is called both in `settle` as in `settle_light`. this seems logical.

however, the name is confusing... what does it mean to _validate an error_? could you please explain what this function aims to do, business logic-wise?

also, some of these parameters don't make sense. the body of the function depends on the conditions of `is_light`, `gas_limit is not None`, and `transaction_error`. if these conditions aren't met, the function will just do nothing.

i suggest we use these conditions to call the method in the first place, and re-word what we are trying to do here.